src/main/scala/beam/agentsim/infrastructure/ChargingFunctions.scala

```diff
@@ -136,11 +138,50 @@ class ChargingFunctions[GEO: GeoLevel](
           origin.time + travelTime1,
           beamVehicle.beamVehicleType
         )
-        val enrouteFactor: Double = (travelTime1 + travelTime2) * inquiry.valueOfTime
-        Map(ParkingMNL.Parameters.EnrouteDetourCost -> enrouteFactor)
-      case _ => Map()
+        (travelTime1 + travelTime2) * inquiry.valueOfTime
+      case _ => 0.0
     }
-    super[ParkingFunctions].setupMNLParameters(parkingAlternative, inquiry) ++ parkingParameters
+
+    // end-of-day parking durations are set to zero, which will be mis-interpreted here
+    val tempParkingDuration = inquiry.searchMode match {
+      case ParkingSearchMode.EnRoute => enrouteDuration.toInt
+      case _                         => inquiry.parkingDuration.toInt
+    }
+    val parkingDuration: Option[Int] =
+      if (tempParkingDuration < estimatedMinParkingDuration)
```

Comment:
This could probably be simplified to:

`val parkingDuration: Option[Int] = Some(Math.max(estimatedMinParkingDuration, tempParkingDuration)`